main.cpp

```diff
@@ -56,14 +56,19 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
+    T(int v, const char* nm)//1
+    {  //2
+        this->value = v;
+        this->name = nm;
+    }   
+    int value;
```

Comment:
When you have to choose where to initialize your member variables and your choices are:
- In-class
- Constructor Member Initializer List
- Inside the Constructor's body
Try to use In-class initialization, followed by the Constructor Member Initializer List.
The reason is that the member variables need to be instantiated/initialized when the class is created.
When you assign to them in the constructor body, you are potentially re-initializing them, which is a waste of resources.
Take a look at the output of this program: https://godbolt.org/z/qj6YMMoYM
Take note of the second instance being created and destroyed due to that constructor body assignment.
- [ ] revise, use the member initializer list to initialize your member variables